flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/BlockingResultInfo.java

```diff
@@ -64,4 +64,7 @@ public interface BlockingResultInfo extends IntermediateResultInfo {
      * @param partitionIndex the intermediate result partition index
      */
     void resetPartitionInfo(int partitionIndex);
+
+    /** Aggregates the subpartition bytes to reduce space usage. */
+    void aggregateSubpartitionBytes();
```

Comment:
No need to add it in this common interface. Instead, in `AdaptiveBatchScheduler#maybeAggregateSubpartitionBytes()` it should invoke this only if it is an `AllToAllBlockingResultInfo`.